src/plugins/intel_gpu/include/intel_gpu/plugin/sync_infer_request.hpp

```diff
@@ -25,8 +25,17 @@ enum class TensorOwner : uint8_t {
 };
 
 struct TensorWrapper {
+    TensorWrapper(const std::shared_ptr<ov::ITensor>& _ptr, TensorOwner _owner)
+        : ptr(_ptr)
+        , owner(_owner)
+        , actual_size(_ptr ? _ptr->get_byte_size() : 0) {}
```

Comment:
Not sure if this is valid situation, but maybe it's better to obtaion original_memory's size instead of tensor's one? Just to prevent the following case:
```
// allocate huge memory buffer
auto output_remote_tensor = gpu_context.create_tensor(output->get_output_element_type(0), allocated_out_shape, shared_output_buffer);

inf_req_shared.set_tensor(output, output_remote_tensor);
inf_req_shared.infer(); // result has tiny shape - OK

inf_req_shared.set_tensor(output, output_remote_tensor);
inf_req_shared.infer(); // result shape is bigger than on previous iteration, but small enough to fit into original memory buffer
// Throw: [GPU] Output tensor set by user has smaller size ...
```

Reply:
We can do that only for usm/remote tensors, but we can't extract allocated mem size from other tensor types, so not sure if different behavior here would be a good thing